hw/display/virtio-dmabuf.c

```diff
@@ -0,0 +1,85 @@
+/*
+ * Virtio Shared dma-buf
+ *
+ * Copyright Red Hat, Inc. 2023
+ *
+ * Authors:
+ *     Albert Esteve <aesteve@redhat.com>
+ *
+ * This work is licensed under the terms of the GNU GPL, version 2 or later.
+ * See the COPYING file in the top-level directory.
+ */
+
+#include "hw/virtio/virtio-dmabuf.h"
+
+
+#define UUID_TO_POINTER(i) \
+    ((gpointer) (g_intern_static_string(qemu_uuid_unparse_strdup((&i)))))
+
+
+static GMutex lock;
+static GHashTable *resource_uuids;
+
+
+static bool virtio_add_resource(QemuUUID uuid, gpointer value)
+{
+    if (resource_uuids == NULL) {
+        resource_uuids = g_hash_table_new_full(NULL, NULL, NULL, g_free);
+    }
+    gpointer struuid = UUID_TO_POINTER(uuid);
+    if (g_hash_table_lookup(resource_uuids, struuid) != NULL) {
```

Comment:
Uhm, the lookup is constant cost and the readability is difficult to measure. For me, those are two different checks: 1) table is not initialized, 2) element already exists. So having them in different ifs makes sense. But since we do not return in the first check, probably can be an `else if` as you suggest. I'll change it and see what qemu maintainers say.